tools/autograd/derivatives.yaml

```diff
@@ -2927,3 +2927,89 @@
 - name: _reshape_copy(Tensor self, SymInt[] size) -> Tensor
   self: grad.reshape_symint(self.sym_sizes())
   result: auto_linear
+
+# TODO(crcrpar): Consider defining fast path with the help of `multi_tensor_apply`.
+# rel: `DERIVATIVE_SINGLE_FOREACH` CodeTemplate.
+- name: _foreach_exp(Tensor[] self) -> Tensor[]
+  self: grads[i] * result[i].conj()
```

Comment:
This formula is a duplicate of the `exp()` function above.
I was thinking that we could extend https://github.com/pytorch/pytorch/blob/f7939b21e1f09521857515f913cd03d13ea2287b/torchgen/api/autograd.py#L325-L360 to re-use that existing formula and put that in the template you already have.

In particular you can redefine the `grad`, `result`, etc that is used in the original formula within the for loop in the template and just dump the formula in there.
WDYT?

Reply:
No you won't need to modify the native functions in any way.
This function is just updating the `formula` field everything else is already properly set.

You just need to generate a good `formula` string for it.

Reply:
I understand it's cleaner by handling things in codegen but
- We could implement a fast path for backward as well as forward.
- with that this PR will cover the whole foreach functions e.g. binary with `Scalar` and `Scalar[]`

Reply:
> We could implement a fast path for backward as well as forward, which would not be compatible with that way of defining backward

You can, this logic that generates the formula kicks in only if you didn't provide a formula already. That's why some inplace functions have formulas and others don't.

> with that this PR will cover the whole foreach functions e.g. binary with Scalar and Scalar[]

Not sure what you mean by that?